### PR TITLE
develop: Enable namespace labels propagation

### PR DESCRIPTION
**Pull request by fmesen**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

These changes allow to enable the propagation of namespace labels into pod logs.

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [X] I have read the [CONTRIBUTING](https://github.com/splunk/splunk-connect-for-kubernetes/blob/develop/CONTRIBUTING.md) doc
- [X] I have read the [CLA](https://github.com/splunk/splunk-connect-for-kubernetes/blob/develop/CLA.md)
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: helm-chart/splunk-connect-for-kubernetes/charts/splunk-kubernetes-logging/values.yaml [hunk 1]
```diff
@@ -58,6 +58,7 @@ k8sMetadata:
     - release
   watch: true
   cache_ttl: 3600
+  propagate_namespace_labels: false
```
Comment (harshit-splunk) [inline helm-chart/splunk-connect-for-kubernetes/charts/splunk-kubernetes-logging/values.yaml:61]: Can you please add a comment to describe what it does?

File: helm-chart/splunk-connect-for-kubernetes/values.yaml [hunk 2]
```diff
@@ -127,6 +127,7 @@ splunk-kubernetes-logging:
       - release
     watch: true
     cache_ttl: 3600
+    propagate_namespace_labels: false
```
Comment (harshit-splunk) [inline helm-chart/splunk-connect-for-kubernetes/values.yaml:130]: Same as above, can you please add a comment to describe what it does?

**Review by harshit-splunk — CHANGES_REQUESTED** [review]
Need to add documentation

File: helm-chart/splunk-connect-for-kubernetes/charts/splunk-kubernetes-logging/values.yaml (hunk 1 above)
Comment (fmesen) [reply helm-chart/splunk-connect-for-kubernetes/charts/splunk-kubernetes-logging/values.yaml:62]: @harshit-splunk The comment has been added.

**Review by harshit-splunk — APPROVED** [review]

**Comment by harshit-splunk** [comment]
Hey @fmesen, Splunk will very soon deprecate `Splunk Connect for Kubernetes` and we recommend moving to [Splunk OpenTelemetry Collector for Kubernetes](https://github.com/signalfx/splunk-otel-collector-chart) which already solves the namespace labels propagation.
You can follow [migration guide](https://github.com/signalfx/splunk-otel-collector-chart/blob/main/docs/migration-from-sck.md).

[Splunk OpenTelemetry Collector for Kubernetes](https://github.com/signalfx/splunk-otel-collector-chart) already supports namespace label propagation with additional feature like regex matching.
Ref: https://github.com/signalfx/splunk-otel-collector-chart/blob/f4cf82bedf200982040e0bb367d58261662b2a2a/helm-charts/splunk-otel-collector/values.yaml#L199-L224

**Pull request merged**